internal/net/conn_other.go

```diff
@@ -0,0 +1,58 @@
+//go:build test
+
+package net
+
+import (
+	"context"
+
+	"github.com/drand/drand/v2/internal/metrics"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
+	insecure "google.golang.org/grpc/credentials/insecure"
+)
+
+// conn retrieve an already existing conn to the given peer or create a new one
+func (g *grpcClient) conn(ctx context.Context, p Peer) (*grpc.ClientConn, error) {
+	g.Lock()
+	defer g.Unlock()
+	var err error
+
+	// we try to retrieve an existing connection if available
+	c, ok := g.conns[p.Address()]
+	if ok && c.GetState() == connectivity.Shutdown {
+		ok = false
+		delete(g.conns, p.Address())
+		metrics.OutgoingConnectionState.WithLabelValues(p.Address()).Set(float64(c.GetState()))
+	}
+
+	// otherwise we try to re-dial it
+	if !ok {
```

Comment:
this whole func would read easier if it was

```
if !ok {

} else if c.getState() == connectivity.Shutdown() {

}
```

Reply:
No because if it's `Shutdown` then we set ok to false and take that `if`

Reply:
ahh i see, ew